### PR TITLE
ct: add `::ContinualTask` to various adapter enums

### PR DESCRIPTION
**Pull request by danhhz**
May as well rip off the band-aid and get it over with.

Touches MaterializeInc/database-issues#8427

### Motivation

  * This PR adds a known-desirable feature.

### Tips for reviewer

This is the first time I've had to make a change to the catalog protos, and I think I missed some steps :)

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/sql/src/session/vars/definitions.rs [hunk 1]
```diff
@@ -541,6 +541,13 @@ pub static MAX_ROLES: VarDefinition = VarDefinition::new(
     true,
 );
 
+pub static MAX_CONTINUAL_TASKS: VarDefinition = VarDefinition::new(
+    "max_continual_tasks",
+    value!(u32; 100),
+    "The maximum number of continual_tasks in the region, across all schemas (Materialize).",
+    true,
```
Comment (danhhz) [inline src/sql/src/session/vars/definitions.rs:548]: This makes the existence of CTs externally visible outside of mz_internal, but that seems better than me forgetting to flip it to public later 🤷 (Also, trying hard to not accumulate more TODO(ct)s, I have plenty)

File: src/catalog/src/builtin.rs [hunk 2]
```diff
@@ -6750,6 +6774,31 @@ ORDER BY 1, 2"#,
     access: vec![PUBLIC_SELECT],
 });
 
+pub static MZ_SHOW_CONTINUAL_TASKS: LazyLock<BuiltinView> = LazyLock::new(|| BuiltinView {
```
Comment (ParkMyCar) [inline src/catalog/src/builtin.rs:6777]: Not required in this PR but we generally index the `SHOW X` commands, [example](https://github.com/MaterializeInc/materialize/blob/main/src/catalog/src/builtin.rs#L7297)

**Review by ParkMyCar — APPROVED** [review]
Nice!!

Not something you need to do in this PR, but to support comments on Continual Tasks we will need to add the new variant of `CommentObjectType` [in planning](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/plan/statement/ddl.rs#L6663).

**Comment by danhhz** [comment]
Sounds good! Given the amount of time I have this week and how likely it is that this will rot, I might pick those two up as TODO(ct)s :). Will add those, rebase, get a nightlies in, and see if I can get this merged

**Comment by danhhz** [comment]
TFTR!

**Pull request merged**